app/src/lib/backend/cloud.ts

```diff
@@ -60,30 +58,66 @@ async function parseResponseJSON(response: Response) {
 	}
 }
 
-interface EvaluatePromptParams {
-	messages: PromptMessage[];
-	temperature?: number;
-	max_tokens?: number;
-	model_kind?: ModelKind;
+export enum RequestMethod {
+	GET = 'GET',
+	POST = 'POST',
+	PUT = 'PUT',
+	PATCH = 'PATCH',
+	DELETE = 'DELETE'
 }
 
+const defaultHeaders = {
+	'Content-Type': 'application/json'
+};
+
 export class CloudClient {
-	fetch: ((input: RequestInfo | URL, init?: RequestInit | undefined) => Promise<Response>) &
-		((input: RequestInfo | URL, init?: RequestInit | undefined) => Promise<Response>);
+	constructor(public fetch = window.fetch) {}
 
-	constructor(fetch: typeof window.fetch = window.fetch) {
-		this.fetch = fetch;
+	private formatBody(body?: FormData | object) {
+		if (body instanceof FormData) {
+			return body;
+		} else if (body) {
+			return JSON.stringify(body);
+		}
+	}
+
+	// TODO: consider renaming
+	async makeRequest<T>(params: {
```

Comment:
I would make this just `request`, and provide convenience functions as necessary e.g. `post`. It makes for nice to read code like `httpClient.post('/some/path', data)`. Wdyt?

Reply:
I've gone with object arguments because I'm not a fan of having multiple optional arguments in a row